File: Rpdb/parse.R
# Benchmark the parsing of a PDB file given as an argument

library(Rpdb)
library(microbenchmark)

pdb_filepath <- commandArgs(trailingOnly=TRUE)[1]

bench <- microbenchmark(read.pdb(pdb_filepath), times=1)

cat(bench$time / 10^9, "\n", sep="")

File: Victor/parse.cc
// Benchmark the parsing of a PDB file given as an argument

#include <PdbLoader.h>
#include <Protein.h>
#include <iostream>
#include <time.h>

using namespace Victor::Biopool;
using namespace Victor;

int main( int argc, char* argv[] ) {
    string pdb_filepath = argv[1];
    clock_t tStart = clock();
    ifstream inFile( pdb_filepath.c_str() );
    // See options at
    // http://protein.bio.unipd.it/victor_doxygen/classVictor_1_1Biopool_1_1PdbLoader.html
    PdbLoader pl(inFile, true, true, false, true, false, false, false, true);
    Protein prot;
    prot.load( pl );
    clock_t tFinish = clock();
    printf("%.6f\n", (double)(tFinish - tStart)/CLOCKS_PER_SEC);
}
